openbas-api/src/main/java/io/openbas/rest/exercise/service/ExerciseService.java

```diff
@@ -431,69 +466,127 @@ private List<ExerciseSimple> execution(TypedQuery<Tuple> query) {
+      return;
+    }
+
+    MappingsByExerciseIds mappingsByExerciseIds =
+        getResultsByExerciseIds(getExerciseIds(exercises));
+
+    for (ExerciseSimple exercise : exercises) {
+      setEmptyGlobalScore(exercise);
+
+      setTargets(exercise, mappingsByExerciseIds);
+    }
+  }
+
+  private static List<ExerciseSimple> getExercisesWithId(List<ExerciseSimple> exercises) {
+    return exercises.stream().filter(exercise -> exercise.getId() != null).toList();
+  }
+
+  private static Set<String> getExerciseIds(List<ExerciseSimple> exercises) {
+    return exercises.stream().map(ExerciseSimple::getId).collect(Collectors.toSet());
+  }
+
+  private MappingsByExerciseIds getResultsByExerciseIds(Set<String> exerciseIds) {
+    Map<String, List<Object[]>> teamsByExerciseIds =
+        getTeamsOrAssetsOrAssetGroupsByExerciseIds(teamRepository.teamsByExerciseIds(exerciseIds));
+
+    Map<String, List<Object[]>> assetsByExerciseIds =
+        getTeamsOrAssetsOrAssetGroupsByExerciseIds(
+            assetRepository.assetsByExerciseIds(exerciseIds));
+
+    Map<String, List<Object[]>> assetGroupByExerciseIds =
+        getTeamsOrAssetsOrAssetGroupsByExerciseIds(
+            assetGroupRepository.assetGroupsByExerciseIds(exerciseIds));
+
+    return new MappingsByExerciseIds(
+        teamsByExerciseIds, assetsByExerciseIds, assetGroupByExerciseIds);
+  }
+
+  private Map<String, List<Object[]>> getTeamsOrAssetsOrAssetGroupsByExerciseIds(
+      List<Object[]> rawTeamsOrAssetsOrAssetGroups) {
+    return ofNullable(rawTeamsOrAssetsOrAssetGroups).orElse(emptyList()).stream()
+        .filter(
+            rawTeamOrAssetOrAssetGroup ->
+                0 < rawTeamOrAssetOrAssetGroup.length && rawTeamOrAssetOrAssetGroup[0] != null)
+        .collect(
+            Collectors.groupingBy(
+                rawTeamOrAssetOrAssetGroup -> (String) rawTeamOrAssetOrAssetGroup[0]));
+  }
+
+  private record MappingsByExerciseIds(
+      Map<String, List<Object[]>> teamsByExerciseIds,
+      Map<String, List<Object[]>> assetsByExerciseIds,
+      Map<String, List<Object[]>> assetGroupsByExerciseIds) {}
+
+  private Map<String, List<RawInjectExpectation>> getExpectationsByExerciseId(
+      Set<String> exerciseIds) {
+    return ofNullable(injectExpectationRepository.rawForComputeGlobalByExerciseIds(exerciseIds))
+        .orElse(emptyList())
+        .stream()
+        .filter(Objects::nonNull)
+        .collect(Collectors.groupingBy(RawInjectExpectation::getExercise_id));
+  }
+
+  private static void setGlobalScore(
+      ExerciseSimple exercise, Map<String, List<RawInjectExpectation>> expectationsByExerciseIds) {
+    exercise.setExpectationResultByTypes(
+        AtomicTestingUtils.getExpectationResultByTypesFromRaw(
+            expectationsByExerciseIds.getOrDefault(exercise.getId(), emptyList())));
+  }
+
+  private static void setEmptyGlobalScore(ExerciseSimple exercise) {
+    exercise.setExpectationResultByTypes(new ArrayList<>());
+  }
```

Comment:
Maybe for this one line, we don't need to extract it into a function.

Reply:
Done